staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -422,3 +422,113 @@ func (rt *debuggingRoundTripper) RoundTrip(req *http.Request) (*http.Response, e
 func (rt *debuggingRoundTripper) WrappedRoundTripper() http.RoundTripper {
 	return rt.delegatedRoundTripper
 }
+
+func legalHeaderByte(b byte) bool {
+	return int(b) < len(legalHeaderKeyBytes) && legalHeaderKeyBytes[b]
+}
+
+func shouldEscape(b byte) bool {
+	// url.PathUnescape() returns an error if any '%' is not followed by two
+	// hexadecimal digits, so we'll intentionally encode it.
+	return !legalHeaderByte(b) || b == '%'
+}
+
+func headerKeyEscape(key string) string {
+	buf := strings.Builder{}
+	for i := 0; i < len(key); i++ {
+		b := key[i]
+		if shouldEscape(b) {
+			// %-encode bytes that should be escaped:
+			// https://tools.ietf.org/html/rfc3986#section-2.1
+			buf.WriteByte('%')
```

Comment:
I'd prefer `fmt.Fprintf(buf, "%%%x", b)` here.

Reply:
In order to ensure a proper encoding, it'd have to be fmt.Fprintf(buf, "%%%x%x", b>>4, b&15):
https://play.golang.org/p/kdKcczb_kTD

FWIW the current scheme is how things are done in the golang stdlib: https://golang.org/src/net/url/url.go?s=7512:7544#L304

Reply:
Try this https://play.golang.org/p/JU_qx6yt-Pw